kratos/geometries/geometry.h

```diff
@@ -3692,8 +3692,8 @@ class Geometry
         double DetJ;
         for ( unsigned int pnt = 0; pnt < integration_points_number; pnt++ )
         {
-            if(rResult[pnt].size1() != this->WorkingSpaceDimension() ||  rResult[pnt].size2() != this->LocalSpaceDimension())
-                rResult[pnt].resize( (*this).size(), this->LocalSpaceDimension(), false );
+            if(rResult[pnt].size1() != (*this).size() ||  rResult[pnt].size2() != this->WorkingSpaceDimension())
+                rResult[pnt].resize( (*this).size(), this->WorkingSpaceDimension(), false );
```

Comment:
```suggestion
            if(rResult[pnt].size1() != (*this).size() ||  rResult[pnt].size2() != this->LocalSpaceDimension())
                rResult[pnt].resize( (*this).size(), this->LocalSpaceDimension(), false );
```
I think that changing `WorkingSpaceDimension` by `size` in the array size1 check is correct. However, as far as I understand, size2 must be the local size. If I'm not wrong changing it to working space dimension would work for "solid"-type geometries but for "beam or shell"-type ones we would be allocating more memory than required, something that might turn into a mismatch in the subsequent array multiplications.

Reply:
I think the result of DN_Dx should be in working space. We already have DN_De which is the gradient of shape functions in local space.

Here this is calculated: DN_Dx = DN_De x JacobianInverse 

In which DN_De has size of (geom.size x LocalSpaceDimension) and JacobianInverse has (LocalSpaceDimension x workingSpace). Therefore the result (DN_Dx) will have the size of (geom.size x workingSpace)

Reply:
I agree with @Vahid-Galavi

Reply:
Maybe I'm missing something but wouldn't this mean that for a line 2D you'll have a gradient X and gradient Y as working space dimension is 2.

Reply:
For a line 2D, in DN_De I expect one derivative, and in DN_Dx I'm expecting two derivatives (X and Y).
Finally, if the line is in a 3D space, then I expect three derivatives in DN_Dx.

Reply:
> Maybe I'm missing something but wouldn't this mean that for a line 2D you'll have a gradient X and gradient Y as working space dimension is 2.

Yes, then the results will be DN_Dx and DN_Dy.

Reply:
> For a line 2D, in DN_De I expect one derivative, and in DN_Dx I'm expecting two derivatives (X and Y). Finally, if the line is in a 3D space, then I expect three derivatives in DN_Dx.

Yes, exactly.